libraries/networking/src/PacketReceiver.h

```diff
@@ -93,4 +135,42 @@ class PacketReceiver : public QObject {
     friend class OctreePacketProcessor;
 };
 
+template <class T>
+PacketReceiver::ListenerReferencePointer PacketReceiver::makeUnsourcedListenerReference(T* target, void (T::* slot)(QSharedPointer<ReceivedMessage>)) {
+    return QSharedPointer<UnsourcedListenerReference<T>>::create(target, slot);
```

Comment:
when I first saw all these shared pointers, I was a little nervous, but under the hood, this code is still doing exactly what the old code was doing, right?  the old code was just doing it in a silly way with strings.  the number of shared pointers + basically all the logic is pretty much staying the same?

Reply:
Not quite sure I understand here, I am likely creating one new shared pointer representing the functor that is being used to call the listener, templated in order to properly use member function references (I believe Qt gets around this with something inside Q_OBJECT), and with two subclassed alternatives to handle the two types of function call parameters.  I attempted to approximate the old code as much as I obviously could, the differences should be about the difference between the two QObject::connect() syntaxes.

Reply:
As I am using subclassed functors they cannot be passed by value (tried that in an earlier version of this code) and pass-by-reference starts running into lifetime issues, thus using shared pointers.